src/index.ts

```diff
@@ -153,6 +153,13 @@ function initConfig(userConfig: Forceable<Config>): TopLevelConfig {
           process.env.GAE_MINOR_VERSION
         ),
       },
+      /**
+       * Our TypeScript interface suggests that only keyFilename and credentials
+       * are accepted, but by passing the entire object to the Trace Writer,
+       * we can allow users to supply other fields that are publicly supported
+       * by the Google Auth Library.
+       */
+      authOptions: mergedConfig,
```

Comment:
Auth options can only come from `userConfig` right? While `mergedConfig` should work, the tradeoff I see is that it may pass too many options and potentially conflict with some other named property in Auth down the road. Passing `userConfig` means that we are implicitly assuming that none of the auth options would be overridden by the environment or default configs.

Reply:
I believe it's plausible that variables in `GCLOUD_TRACE_CONFIG` can be used for authentication. At least, that was the behavior from before.